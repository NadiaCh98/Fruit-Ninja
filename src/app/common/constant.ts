export const SCENE_SIZE = 9;
