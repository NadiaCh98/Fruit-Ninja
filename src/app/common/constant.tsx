import { StartMenuItem } from '../game/models/startMenu';
import { GameConfig } from '../game/models/game';
import { ReactComponent as ExitIcon } from '../../assets/icons/close.svg';
import { ReactComponent as PauseIcon } from '../../assets/icons/pause.svg';
import { ReactComponent as ReplayIcon } from '../../assets/icons/replay.svg';
import { ReactComponent as PlayIcon } from '../../assets/icons/play.svg';

export const SCENE_SIZE = 9;
export const ATTEMPS = 3;
export const MIN_CUT_COMBO = 3;
export const DELAY_BETWEEN_FRUITS_GROUP = 3000;

export enum Fruits {
  apple,
  banana,
  lemon,
  pear
}

export enum FruitSpeed {
  Low = 2.8,
  Average = 3,
  Fast = 3.4,
}

export enum Game {
  Classic = 'Classic',
  Dzen = 'Dzen',
}

export const START_MENU: StartMenuItem[] = [
  {
    mode: Game.Classic,
    fruit: 'apple',
  },
  {
    mode: Game.Dzen,
    fruit: 'apple',
  },
];

export const GAME_CONFIG: GameConfig[] = [
  {
    game: Game.Classic,
    timer: null,
    attempts: 3,
  },
  {
    game: Game.Dzen,
    timer: 90000,
    attempts: 0,
  },
];

export enum PermissibleButton {
  Exit = 'Exit',
  Pause = 'Pause',
  Play = 'Play',
  Replay = 'Replay',
}

export const BUTTONS: Record<PermissibleButton, JSX.Element> = {
  Exit: <ExitIcon />,
  Pause: <PauseIcon />,
  Play: <PlayIcon />,
  Replay: <ReplayIcon />,
};

export const RIGHT_POSITION: React.CSSProperties = {
  top: 0,
  right: 0,
  width: 'max-content',
};
