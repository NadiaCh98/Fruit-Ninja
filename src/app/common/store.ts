import { GameStore } from '../gameStore';
import { FRUITS_POSITION_INTERVAL } from './constant';
import { createContext } from './storeUtils';

export const { StoreProvider, useStore } = createContext({
  Game: new GameStore(FRUITS_POSITION_INTERVAL),
});
