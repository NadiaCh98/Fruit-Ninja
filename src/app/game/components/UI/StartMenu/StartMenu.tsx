import React from 'react';
import { Button } from '../../../../common/components/Button/Button';
import { GameMode } from '../../../../common/constant';
import { StartMenuItem } from '../../../models/startMenu';
import { Modal } from '../../../../common/components/Modal/Modal';
import styled from 'styled-components';

interface StartMenuProps {
  readonly items: StartMenuItem[];
  readonly selectMode: (mode: GameMode) => void;
}

const StyledMenu = styled.div`
  display: grid;
  grid-gap: 10px;
  width: 20rem;
`;

const startGameSound = new Audio('/sound/game-start.wav');

export const StartMenu: React.FC<StartMenuProps> = ({ items, selectMode }) => {
  const selectGameMode = React.useCallback(
    (mode: GameMode) => () => {
      startGameSound.play();
      selectMode(mode);
    },
    [selectMode]
  );

  return (
    <Modal>
      <StyledMenu data-testid="modes">
        {items.map(({ mode }, i) => (
          <Button key={i} kind={'primary'} clickHandler={selectGameMode(mode)}>
            <p>{mode}</p>
          </Button>
        ))}
      </StyledMenu>
    </Modal>
  );
};
