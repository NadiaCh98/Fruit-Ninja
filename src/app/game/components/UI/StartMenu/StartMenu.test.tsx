import { fireEvent, render } from '@testing-library/react';
import React from 'react';
import { START_MENU } from '../../../../common/constant';
import { StartMenu } from './StartMenu';

describe('Start Menu', () => {
  const selectMode = jest.fn();
  const startMenu = <StartMenu selectMode={selectMode} items={START_MENU} />;

  it('has a correct template', () => {
    const { container } = render(startMenu);

    expect(container).toMatchSnapshot();
  });

  it('asserts that selectMode() is fired with correct mode', () => {
    const { getByTestId } = render(startMenu);

    fireEvent.click(getByTestId('modes').childNodes[0]);

    expect(selectMode).toBeCalledTimes(1);
    expect(selectMode).toBeCalledWith(START_MENU[0].mode);
  });
});
