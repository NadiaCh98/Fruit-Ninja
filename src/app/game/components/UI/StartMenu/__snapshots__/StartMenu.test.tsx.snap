// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Start Menu has a correct template 1`] = `
<div>
  <div
    class="sc-gtsrHT dVXCLS"
  />
  <div
    class="sc-dlnjwi glLXEG"
  >
    <div
      class="sc-hKFxyN cStBoB"
      data-testid="modes"
    >
      <button
        class="sc-bdnxRM mZLZO"
        kind="primary"
      >
        <p>
          Classic
        </p>
      </button>
      <button
        class="sc-bdnxRM mZLZO"
        kind="primary"
      >
        <p>
          Dzen
        </p>
      </button>
    </div>
  </div>
</div>
`;
